CMakeLists.txt

```diff
@@ -0,0 +1,934 @@
+endif()
+
+set(git_builtin_extra
+cherry cherry-pick format-patch fsck-objects
+init merge-subtree restore show
+stage status switch whatchanged)
+
+#Creating symlinks
+
+foreach(s ${git_SOURCES} ${git_builtin_extra})
+	string(REPLACE "builtin/" "" s ${s})
+	string(REPLACE ".c" "" s ${s})
+	add_custom_command(OUTPUT git-${s}${EXE}
+						COMMAND ${CMAKE_COMMAND} -E create_symlink ${CMAKE_BINARY_DIR}/git${EXE} ${CMAKE_BINARY_DIR}/git-${s}${EXE}
+						DEPENDS git)
+	list(APPEND git_symlinks ${CMAKE_BINARY_DIR}/git-${s}${EXE})
+endforeach()
+
+if(CURL_FOUND)
+	set(remote_exes 
+		git-remote-https git-remote-ftp git-remote-ftps)
+	foreach(s ${remote_exes})
+		add_custom_command(OUTPUT ${s}${EXE}
+						COMMAND ${CMAKE_COMMAND} -E create_symlink ${CMAKE_BINARY_DIR}/git-remote-http${EXE} ${CMAKE_BINARY_DIR}/${s}${EXE}
+						DEPENDS git-remote-http)
+		list(APPEND git_http_symlinks ${CMAKE_BINARY_DIR}/${s}${EXE})
+	endforeach()
+endif()
+
+add_custom_target(git-symlinks ALL DEPENDS ${git_symlinks} ${git_http_symlinks})
+
+#creating required scripts
+if(WIN32)
+	set(LOCALEDIR /mingw64/share/locale)
+	set(GITWEBDIR /mingw64/share/locale)
+	set(INSTLIBDIR /mingw64/share/perl5)
+else()
+	set(LOCALEDIR /home/$ENV{USER}/share/locale)
+	set(GITWEBDIR /home/$ENV{USER}/share/locale)
+	set(INSTLIBDIR /home/$ENV{USER}/share/perl5)
+endif()
+
+set(git_shell_scripts
+	git-bisect git-difftool--helper git-filter-branch
+	git-merge-octopus git-merge-one-file git-merge-resolve
+	git-mergetool git-quiltimport git-legacy-stash
+	git-request-pull git-submodule git-web--browse
+	git-mergetool--lib git-parse-remote git-rebase--preserve-merges
+	git-sh-setup git-sh-i18n git-instaweb)
+
+foreach(scr ${git_shell_scripts})
+	file(STRINGS ${CMAKE_SOURCE_DIR}/${scr}.sh content NEWLINE_CONSUME)
+	string(REPLACE "@SHELL_PATH@" "/bin/sh" content "${content}")
+	string(REPLACE "@@DIFF@@" "diff" content "${content}")
+	string(REPLACE "@LOCALEDIR@" "${LOCALEDIR}" content "${content}")
+	string(REPLACE "@GITWEBDIR@" "${GITWEBDIR}" content "${content}")
+	string(REPLACE "@@NO_CURL@@" "" content "${content}")
+	string(REPLACE "@@USE_GETTEXT_SCHEME@@" "" content "${content}")
+	string(REPLACE "# @@BROKEN_PATH_FIX@@" "" content "${content}")
+	string(REPLACE "@@PERL@@" "/usr/bin/perl" content "${content}")
+	string(REPLACE "@@SANE_TEXT_GREP@@" "-a" content "${content}")
+	string(REPLACE "@@PAGER_ENV@@" "LESS=FRX LV=-c" content "${content}")
+	file(WRITE ${CMAKE_BINARY_DIR}/${scr} ${content})
+endforeach()
+
+set(git_perl_scripts
+git-add--interactive git-archimport git-cvsexportcommit 
+git-cvsimport git-cvsserver git-send-email git-svn) 
+
+
+file(STRINGS ${CMAKE_SOURCE_DIR}/perl/header_templates/fixed_prefix.template.pl perl_header )
+string(REPLACE "@@PATHSEP@@" ":" perl_header "${perl_header}")
+string(REPLACE "@@INSTLIBDIR@@" "${INSTLIBDIR}" perl_header "${perl_header}")
+#message(${perl_header})
+
+foreach(scr ${git_perl_scripts})
```

Comment:
Maybe spell out `script` (instead of shortening it to `scr`)? This applies elsewhere, too.

Reply:
okay